subiquity/client/controllers/ubuntu_advantage.py

```diff
@@ -44,21 +37,6 @@ class UbuntuAdvantageController(SubiquityTuiController):
 
     endpoint_name = "ubuntu_advantage"
 
-    def __init__(self, app):
-        """ Initializer for client-side UA controller. """
-        strategy: UAInterfaceStrategy
-        if app.opts.dry_run:
-            strategy = MockedUAInterfaceStrategy(scale_factor=app.scale_factor)
-        else:
-            # Make sure we execute `$PYTHON "$SNAP/usr/bin/ubuntu-advantage"`.
-            executable = (
-                os.environ["PYTHON"],
-                os.path.join(os.environ["SNAP"], "usr/bin/ubuntu-advantage"),
-            )
-            strategy = UAClientUAInterfaceStrategy(executable=executable)
-        self.ua_interface = UAInterface(strategy)
-        super().__init__(app)
-
     async def make_ui(self) -> UbuntuAdvantageView:
         """ Generate the UI, based on the data provided by the model. """
```

Comment:
github won't let me comment on the relevant code, but the decision on whether or not to skip this page should probably be more server-side too (fine for a follow up branch!)

Reply:
Thanks! I'll go for the follow-up branch :)

I think it would make sense to have a generic way to skip a page based on the server controller's decision. Maybe by marking the controller as non-interactive if we can?

Reply:
Yeah, overriding interactive() is the cleanest and most generic way to make the client skip a page. Maybe we should support raising a particular exception from a GET method as well to trigger the same behaviour, that would be a fairly easy change in the middleware.